test/test-parse-warning.cpp

```diff
@@ -774,7 +774,7 @@ namespace quick_lint_js
             }));
```

Comment:
Must fix: This test is failing on my machine:
```
/Users/strager/Projects/quick-lint-js/test/test-parse-warning.cpp:774: Failure
Value of: p.errors
Expected: has 1 element that has type diag_mistyped_strict_inequality_operator
  Actual: {}
```
I think it's failing because you're parsing an `if` statement with `parse_and_visit_expression`, but `parse_and_visit_statement` is the function you should be calling.

Reply:
I will fix this and see if the  tests pass, if so I think we're golden